src/components/QueryEditor/MetricAggregationsEditor/state/reducer.ts

```diff
@@ -32,31 +33,11 @@ export const reducer = (state: OpenSearchQuery['metrics'], action: Action): Open
   }
 
   if (changeMetricType.match(action)) {
-    return state
-      ?.filter((metric) =>
-        // When the new metric type is `isSingleMetric` we remove all other metrics from the query
-        // leaving only the current one.
-        !!metricAggregationConfig[action.payload.type].isSingleMetric ? metric.id === action.payload.id : true
-      )
-      .map((metric) => {
-        if (metric.id !== action.payload.id) {
-          return metric;
-        }
-
-        /*
-        TODO: The previous version of the query editor was keeping some of the old metric's configurations
-        in the new selected one (such as field or some settings).
-        It the future would be nice to have the same behavior but it's hard without a proper definition,
-        as Elasticsearch will error sometimes if some settings are not compatible.
-      */
-        return {
-          id: metric.id,
-          type: action.payload.type,
-          ...metricAggregationConfig[action.payload.type].defaults,
-        } as MetricAggregation;
-      });
+    return getNewMetrics(state || [], action.payload.id, action.payload.type);
+  }
+  if (updateLuceneTypeAndMetrics.match(action)) {
+    return action.payload.metrics;
```

Comment:
nit: from a code organization perspective, it feels a little weird that `changeMetricType` calls `getNewMetrics` here and `updateLuceneTypeAndMetrics` calls it in the `LuceneQueryTypeSelector` when it dispatches the action

Reply:
Updated to send metric type and id instead of metrics, so just reducer will call `getNewMetrics`